### PR TITLE
QA work for null states

### LINKED ISSUE #64: [Shows] Don’t show ‘Current & Upcoming’ shows when there are none.
E.g. https://www.artsy.net/artist/peter-doig-2/shows

### LINKED ISSUE #65: [Shows] Don’t try to show data that’s not available
E.g. https://www.artsy.net/artist/peter-doig-2/shows has a past show where the location information is missing.

### LINKED ISSUE #66: [Biography] Don’t try to remove Markdown when there is no bio.
E.g. https://www.artsy.net/artist/zhang-hongtu

### LINKED ISSUE #67: [Articles] Don’t show ‘Featured Articles’ when there are none.
E.g. https://www.artsy.net/artist/zhang-hongtu

### LINKED ISSUE #68: [Related Artists] Use singular ‘work’ if an artist only has 1 work.
E.g. https://www.artsy.net/artist/rob-pruitt has a link to ‘Kaikai Kiki’ which has only 1 work:

img_7791

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Got show images on iPad to update width properly
- [x] Hides blurb if there is no blurb
- [x] Hides Current & Upcoming Shows if there are none
- [x] Uses 'work' instead of 'works' for singular works in related artist component
- [x] Hides articles if there are no articles for artist
- [x] Hide show data if missing
- [x] Hide ‘Other Works’ if all works are for sale. (see https://www.artsy.net/artist/nicole-cohen/works)

simulator screen shot may 23 2016 1 13 36 pm
simulator screen shot may 23 2016 1 14 09 pm

Fixes #68.
Fixes #67.
Fixes #66.
Fixes #64.
Will fix #65.

### CODE REVIEW

File: lib/components/artist/shows/metadata.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Metadata extends React.Component {
         <Text style={styles.sansSerifText}>{this.props.show.partner.name.toUpperCase()}</Text>
         <Text style={styles.sansSerifText}>{this.showTypeString(this.props.show)}</Text>
         <SerifText style={styles.serifText}>{this.props.show.name}</SerifText>
-        { this.dateAndLocationString(this.props.show) }
+        { this.props.show.location ? this.dateAndLocationString() : null }
```
Comment (alloy) [inline lib/components/artist/shows/metadata.js]: Why not perform the location null check in that method as well? Probably reads better to have all conditionals in one place.

**Comment by alloy** [comment]
Added another TODO for when all works are for sale.

**Comment by sarahscott** [comment]
I'm going to make a separate PR for the images on iPad because that's going to be its own investigation. Also, @briansw , what should we display if an artist has no shows (see https://www.artsy.net/artist/kaikai-kiki)?

**Comment by briansw** [comment]
@sarahscott is it simple enough to hide the shows tab in that case?

**Comment by sarahscott** [comment]
@briansw  I'll look into it. @alloy might know better.

**Comment by alloy** [comment]
@sarahscott @briansw I don’t think that should be a problem, you have those counts before any view is rendered, so you should be able to use that to determine what tabs to render.

**Comment by alloy** [comment]
Something like:

```js
const tabs = ['ABOUT', 'WORKS'];
if (this.artist.previous_shows.length && this.artist.upcoming_shows.length && this.current_shows) {
  tabs.push('SHOWS');
}
…
<TabView titles={tabs}>
  …
</TabView>
```

You’ll need to pull the show fragments that define the previous, upcoming, and current shows lists from their respective components into the Artist component, otherwise you can’t access them.

File: lib/components/artist/related_artists/related_artist.js [hunk 2]
```diff
@@ -15,13 +15,13 @@ const imageWidth = windowDimensions.width > 700 ? (windowDimensions.width - 100)
...
   handleTap() {
-    SwitchBoard.presentNavigationViewController(this, `/artist/${this.props.artist.href}`);
+    SwitchBoard.presentNavigationViewController(this, this.props.artist.href);
   }
 
   render() {
     const artist = this.props.artist;
     return (
-      <TouchableWithoutFeedback onPress={this.handleTap.bind(this)}>
+      <TouchableWithoutFeedback onPress={this.handleTap.bind(this)} delayPressIn={5} delayPressOut={5}>
```
Comment (alloy) [inline lib/components/artist/related_artists/related_artist.js]: So does this help or is this left-over experimenting?

**Comment by alloy** [comment]
Merging including those tap delays, but please let me know what the deal is with them.

File: lib/components/artist/related_artists/related_artist.js (hunk 2 above)
Comment (sarahscott) [inline lib/components/artist/related_artists/related_artist.js]: Woops - yes it's a left-over experiment. Will remove in my next pr.

**Comment by alloy** [comment]
🎉 

Great work, @sarahscott :clap:

**Pull request merged**